src/conn/conn_api.c

```diff
@@ -2077,7 +2077,7 @@ __wt_extra_diagnostics_config(WT_SESSION_IMPL *session, const char *cfg[])
     flags = 0;
     for (ft = extra_diagnostics_types; ft->name != NULL; ft++) {
         if ((ret = __wt_config_subgets(session, &cval, ft->name, &sval)) == 0 && sval.val != 0)
-            LF_SET(ft->flag);
+            LF_SET((uint16_t)ft->flag);
```

Comment:
I can't say I like the repeated casting but this `WT_NAME_FLAG`, is kinda weird and frequently `or`ing together 64 bits with a different amount of bits. We could make all the `conn` flags 64 bits but i'm guessing this isn't the only place we'll need those casts? 

I thought we discussed this however and the idea of a `F_SET_16` etc came up but it turned out we only needed to fix windows? Can you clarify.

Reply:
This seems much better.